File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        right=len(height)-1
        left=0
        maxi=0
        while left<=right:
            w=right-left
            a=min(height[right],height[left])*w
            maxi=max(maxi,a)
            if height[left]<height[right]:
                left+=1
            else:
                right-=1
        return maxi

File: 0011-container-with-most-water/0011-container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left=0;
        int n=height.Length-1;
        int right=n;
        int maxi=0;
        int ans=0;
        while(left<right){
            int mx=Math.Min(height[left],height[right])*(right-left);
            ans=Math.Max(ans,mx);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }

        }
        return ans;
    }
}